src/components/v5/common/Fields/CardSelect/CardSelect.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useCallback, useMemo, useState } from 'react';
+
+import clsx from 'clsx';
+import { useIntl } from 'react-intl';
+import useToggle from '~hooks/useToggle';
+import Card from '~v5/shared/Card';
+import { CardSelectProps } from './types';
+import { FIELD_STATE } from '../consts';
+
+const displayName = 'v5.common.Fields.CardSelect';
+
+function CardSelect<TValue = string>({
+  // eslint-disable-next-line no-warning-comments
+  // TODO: add support for message when needed
+  // message,
+  options,
+  onChange: onChangeProp,
+  value: valueProp,
+  title,
+  keyExtractor = (v) => String(v),
+  valueComparator = (v1, v2) => v1 === v2,
+  state,
+  placeholder,
+}: CardSelectProps<TValue>): JSX.Element {
+  const intl = useIntl();
+
+  const [
+    isSelectVisible,
+    { toggle: toggleSelect, toggleOff: toggleSelectOff, registerContainerRef },
+  ] = useToggle();
+  const [uncontrolledValue, setUncontrolledValue] = useState<TValue>();
+  const uncontrolledOnChange = useCallback((newValue: TValue) => {
+    setUncontrolledValue(newValue);
+  }, []);
+
+  const value = valueProp ?? uncontrolledValue;
+  const onChange = onChangeProp ?? uncontrolledOnChange;
+
+  const selectedOption = useMemo(() => {
+    if (value === undefined) {
```

Comment:
Can't this be `if (!value)`?

Reply:
It will fail if `TValue` will be number and one of options value would be for example `0`